src/sentry/sentry_apps/api/endpoints/sentry_app_webhook_requests.py

```diff
@@ -0,0 +1,149 @@
+from datetime import datetime, timezone
+
+from dateutil.parser import parse as parse_date
+from rest_framework import serializers, status
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from sentry.api.api_owners import ApiOwner
+from sentry.api.api_publish_status import ApiPublishStatus
+from sentry.api.base import control_silo_endpoint
+from sentry.api.serializers import serialize
+from sentry.models.organizationmapping import OrganizationMapping
+from sentry.sentry_apps.api.bases.sentryapps import SentryAppBaseEndpoint, SentryAppStatsPermission
+from sentry.sentry_apps.api.serializers.sentry_app_webhook_request import (
+    SentryAppWebhookRequestSerializer,
+)
+from sentry.sentry_apps.api.utils.webhook_requests import (
+    BufferedRequest,
+    get_buffer_requests_from_control,
+    get_buffer_requests_from_regions,
+)
+from sentry.sentry_apps.models.sentry_app import SentryApp
+from sentry.sentry_apps.services.app_request import RpcSentryAppRequest, SentryAppRequestFilterArgs
+from sentry.utils.sentry_apps import EXTENDED_VALID_EVENTS
+
+
+class IncomingRequestSerializer(serializers.Serializer):
+    date_format = "%Y-%m-%d %H:%M:%S"
+    eventType = serializers.ChoiceField(
+        choices=EXTENDED_VALID_EVENTS,
+        required=False,
+    )
+    errorsOnly = serializers.BooleanField(default=False, required=False)
+    organizationSlug = serializers.CharField(required=False)
+    start = serializers.DateTimeField(
+        format=date_format,
+        default=datetime.strptime("2000-01-01 00:00:00", date_format).replace(tzinfo=timezone.utc),
+        default_timezone=timezone.utc,
+        required=False,
+    )
+    end = serializers.DateTimeField(
+        format=date_format, default=None, default_timezone=timezone.utc, required=False
+    )
+
+    def validate(self, data):
+        if "start" in data and "end" in data and data["start"] > data["end"]:
+            raise serializers.ValidationError("Invalid timestamp (start must be before end).")
+        return data
+
+    def validate_end(self, end):
+        if end is None:
+            end = datetime.now(tz=timezone.utc)
```

Comment:
i am not 100% confidence w time zones, but would this work if someone gives no end but their timezone is past utc?

Reply:
i thinkkk this should be fine? all of the datetimes are in utc, it doesn't matter what time zone the user is in.

Reply:
yeah ur right, i thought about this more and it should just update the current time to be in utc